bag.go

```diff
@@ -32,6 +32,12 @@ func New() *ParameterBag {
 
 }
 
+//ParameterBagConstructor can be registered with the  bag factory
+func ParameterBagConstructor() Bag {
```

Comment:
You are already in parameterbag. Maybe `parameterbag.ParameterBagConstructor` is a bit verbose? 
Maybe we can find a better naming